### PR TITLE
[EC-222] Add GitHub repository settings in a Terraform configuration

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

Define this GitHub repository settings through a Terraform configuration

### Motivation and context

IaC is better than UI

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [ ] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### Other information

---

### CODE REVIEW

File: infra/repository/github_environment_cd.tf [hunk 1]
```diff
@@ -0,0 +1,18 @@
+resource "github_repository_environment" "github_repository_environment_prod_cd" {
+  environment = "prod-cd"
```
Comment (lucacavallaro) [inline infra/repository/github_environment_cd.tf:2]: what is the purpose of this environment?

File: infra/repository/github_environment_ci.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+resource "github_repository_environment" "github_repository_environment_prod_ci" {
+  environment = "prod-ci"
```
Comment (lucacavallaro) [inline infra/repository/github_environment_ci.tf:2]: what is the purpose of this environment?

File: infra/repository/github_environment_cd.tf (hunk 1 above)
Comment (Krusty93) [reply infra/repository/github_environment_cd.tf:2]: To manage azure login through federated credentials and deployment reviewers

File: infra/repository/github_environment_ci.tf (hunk 2 above)
Comment (Krusty93) [reply infra/repository/github_environment_ci.tf:2]: To manage azure login through federated credentials and deployment reviewers

File: infra/repository/github_environment_cd.tf (hunk 1 above)
Comment (lucacavallaro) [reply infra/repository/github_environment_cd.tf:2]: ok, but what are the differences between `prod-cd` and `prod-cd`?

**Pull request merged**